internal/wasm/memory.go

```diff
@@ -294,6 +294,15 @@ func PagesToUnitOfBytes(pages uint32) string {
 
 // Below are raw functions used to implement the api.Memory API:
 
+// Uses atomic write to update the length of a slice.
+func atomicStoreLengthAndCap(slice *[]byte, length uintptr, cap uintptr) {
```

Comment:
Somehow duplicating with `atomicStoreLength` seemed cleanest of what I could come up with but happy to take suggestions

Reply:
Not sure if they both need to be atomic.
But if we're going to use atomic, which doesn't hurt, I'd switch order: first update `cap` then `len`.
This ensures it's impossible to observe `len>cap`.

Reply:
Ah makes sense